src/screens/ProfileScreen/ProfileScreen.js

```diff
@@ -1,250 +1,290 @@
-                setOccupation(doc.data().occupation)
-                setDisplayImage(doc.data().downloadURL)
-                setProfilePresent(doc.data().isImagePresent)
-            } else {
-              alert("No Document Found")
-            }
-        }).catch((error) => {
-            alert(error)
-        });     
-  }, [])
-
-    const uploadImage = async () => {
-      setProgressUpload(false)
-      if(image == null){
-        savePostData(displayImage) 
-        return
-      }
-        const uri = image;
-        const childPath = `post/${firebase.auth().currentUser.uid}/profile_picture.png`;
-
-        const response = await fetch(uri);
-        const blob = await response.blob();
-
-        const task = firebase
-            .storage()
-            .ref()
-            .child(childPath)
-            .put(blob);
-
-        const taskCompleted = () => {
-            task.snapshot.ref.getDownloadURL().then((snapshot) => {
-                savePostData(snapshot);
-            }).catch((error) => {
-              alert("Please Upload all the Information below")
-            })
-        }
+  const userID1 = firebase.auth().currentUser.uid;
+  const ref = firebase.firestore().collection('users').doc(userID1);
 
-        const taskError = snapshot => {
-            alert("Error Occured")
+  useEffect(() => {
+    ref
+      .get()
+      .then(doc => {
+        if (doc.exists) {
+          setName(doc.data().fullName);
+          setInterest(doc.data().interest);
+          setCountry(doc.data().country);
+          setOccupation(doc.data().occupation);
+          setDisplayImage(doc.data().downloadURL);
+          setProfilePresent(doc.data().isImagePresent);
+        } else {
+          alert('No Document Found');
         }
+      })
+      .catch(error => {
+        alert(error);
+      });
+  }, []);
 
-        task.on("state_changed", snapshot => {
-          setTransferred(
-            Math.round(snapshot.bytesTransferred / snapshot.totalBytes) * 10000
-          );
-        }, taskError, taskCompleted);   
-    }
-    
-    const savePostData = (downloadURL) => {
-      setEditfield(false)
-      setDisplayImage(downloadURL)
-        ref.update({
-                country,
-                occupation,
-                interest,
-                downloadURL,
-                creation: firebase.firestore.FieldValue.serverTimestamp(),
-                isImagePresent: true,
-            }).then(() => {
-                alert("Updated Profile")
-                setImageUpload(true)
-                setProgressUpload(true)
-                setTransferred(0)
-            }).catch((error) => {
-              alert(error)
-         })
+  const uploadImage = async () => {
+    setProgressUpload(false);
+    if (image == null) {
+      savePostData(displayImage);
+      return;
     }
+    const uri = image;
+    const childPath = `post/${
+      firebase.auth().currentUser.uid
+    }/profile_picture.png`;
```

Comment:
If this 'profile_picture.png' string is having multiple occurrences on this same file, replace it with a constant. otherwise please ignore this comment.

Reply:
profile_picture.png is only used once for the path only